nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/ScriptUtils.java

```diff
@@ -46,23 +44,19 @@
 import javax.script.ScriptEngineManager;
 import javax.script.ScriptException;
 
-import org.apache.nifi.annotation.lifecycle.OnStopped;
 import org.apache.nifi.components.AllowableValue;
 import org.apache.nifi.components.PropertyDescriptor;
 import org.apache.nifi.components.ValidationContext;
 import org.apache.nifi.components.ValidationResult;
 import org.apache.nifi.components.Validator;
-import org.apache.nifi.processor.AbstractSessionFactoryProcessor;
 import org.apache.nifi.processor.Relationship;
 import org.apache.nifi.processor.util.StandardValidators;
 import org.apache.nifi.util.StringUtils;
 
 /**
- * This class contains variables and methods common to scripting processors
+ * This class contains variables and methods common to scripting processors, reporting tasks, etc.
  */
-@Stateful(scopes = {Scope.LOCAL, Scope.CLUSTER},
-        description = "Scripts can store and retrieve state using the State Management APIs. Consult the State Manager section of the Developer's Guide for more details.")
-public abstract class AbstractScriptProcessor extends AbstractSessionFactoryProcessor {
+public class ScriptUtils {
```

Comment:
I agree the OO/structure could definitely be better. Before I added my previous comments I had tried breaking up the utilities and constants, but it still seemed messy with the interactions between the two classes (and the processors), if the static property descriptors were in a separate class but their usage was in the utils class (and the processors themselves) directly from the constants class, when in real OO they should be passed in as arguments and such.
Since only a very few people will know enough about the history of the code to deduce that the bad OO is a result of not having multiple inheritance, your point about maintainable code is well-taken. I will refactor it based on your suggestions, and then just test the damn out of it to make sure there are no regressions :)